components/radial-chart/src/radial-chart.ts

```diff
@@ -0,0 +1,293 @@
+    }
+  }
+  private _type: 'pie' | 'donut' = 'pie';
+
+  /** Maximum value the chart displays, 100%, full circle */
+  @Input()
+  get maxValue(): number | null {
+    return this._maxValue;
+  }
+  set maxValue(value: number | null) {
+    this._maxValue = coerceNumberProperty(value);
+    this._updateRenderData();
+  }
+  private _maxValue: number | null = null;
+
+  /** Where the chart's legend should be placed */
+  @Input()
+  get legendPosition(): 'right' | 'bottom' {
+    return this._legendPosition;
+  }
+  set legendPosition(value: 'right' | 'bottom') {
+    this._legendPosition = value;
+  }
+  private _legendPosition: 'right' | 'bottom' = 'right';
+
+  /** @internal Series data, <dt-radial-chart-series> */
+  @ContentChildren(DtRadialChartSeries) _radialChartSeries: QueryList<
+    DtRadialChartSeries
+  >;
+
+  /** @internal Overlay template */
+  @ContentChild(DtRadialChartOverlay, { static: false, read: TemplateRef })
+  _overlay: TemplateRef<{ $implicit: DtRadialChartOverlayData }>;
+
+  /** @internal Render data used to draw the SVG. */
+  _renderData: DtRadialChartRenderData[] = [];
+
+  /** @internal Destroy subject to clear subscriptions on component destroy. */
+  private readonly _destroy = new Subject<void>();
+
+  /** @internal The chart's width based on the viewport width. */
+  _width = 0;
+
+  /** @internal The chart's radius. */
+  get _radius(): number {
+    // tslint:disable-next-line: no-magic-numbers
+    return this._width / 2;
+  }
+
+  /**
+   * @internal
+   * Flag if a background shape should be rendered.
+   * Depends on difference of the given and the calculated max value.
+   */
+  _hasBackground = false;
+
+  /**
+   * @internal
+   * The background path is either a full circle or a ring.
+   * Depends on the chart type.
+   */
+  _backgroundPath: string;
+
+  /**
+   * @internal
+   * Sum of all series values.
+   */
+  _totalSeriesValue: number;
+
+  /**
+   * @internal
+   * Get viewbox data based on the radius and width.
+   * Adjust the viewbox to center the circle.
+   */
+  get _viewBox(): string {
+    // tslint:disable-next-line: no-magic-numbers
+    return `${-this._width / 2} ${-this._width / 2} ${this._width} ${
+      this._width
+    }`;
+  }
+
+  /** @internal The chart's inner radius based on the chart type and defined fraction. */
+  get _innerRadius(): number {
+    return this.type === 'pie' ? 0 : this._radius * DONUT_INNER_CIRCLE_FRACTION;
+  }
+
+  constructor(
+    private _elementRef: ElementRef<HTMLElement>,
+    private _viewportResizer: DtViewportResizer,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _sanitizer: DomSanitizer,
+  ) {}
+
+  /** AfterContentInit hook */
+  ngAfterContentInit(): void {
+    /**
+     * Fires every time a value within one of the series changes
+     * or the series data itself gets an update (one series is added or removed).
+     */
+    this._radialChartSeries.changes
+      .pipe(
+        takeUntil(this._destroy),
```

Comment:
As I have learned from @lukasholzer, the `takeUntil` should be last in the pipeline.